src/cunumeric/index/wrap_template.inl

```diff
@@ -60,17 +60,37 @@ struct WrapImpl {
     assert(volume_in != 0);
 #endif
 
-    WrapImplBody<KIND, DIM>()(out, pitches_out, out_rect, pitches_in, input_rect, dense);
+    if (args.has_input) {
+      auto in_rect = args.in.shape<1>();
+      auto in = args.in.read_accessor<int64_t, 1>(in_rect);  // input should be always integer type
+#ifdef DEBUG_CUNUMERIC
+      assert(in_rect == out_rect);
+#endif
+#ifndef LEGION_BOUNDS_CHECKS
+      dense = dense && in.accessor.is_dense_row_major(out_rect);
+#endif
```

Comment:
It doesn't matter whether the `in` store is dense, since you always access that through the accessor interface (which works whether dense or not). It only matters for `out`, which you access directly through the pointer (if dense).

Reply:
right.